folly/detail/TypeList.h

```diff
@@ -168,16 +168,18 @@ using If = MetaApply<impl::If_<If_>, Then, Else>;
  */
 template <template <class...> class C, class... Ts>
 class MetaDefer {
-  template <template <class...> class D = C, class = D<Ts...>>
+  template <template <class...> class D, class = D<Ts...>>
```

Comment:
Does this need to be `void_t<D<Ts...>>` (folly's `void_t`)?

Reply:
Hmm I don't think it _needs_ to be: if `D<Ts...>` is ill-formed it will cause a substitution failure and overload resolution picks `try_(long)`, else they're both valid so `try_(int)` gets called? But maybe since I'm using `void` as the default parameter on `try_(long)` it would be a clarity improvement, so I'm happy to write it that way if you'd prefer.

Reply:
Can all of the changes in this file be reverted given the addition of `PolySelf_`?

Reply:
Unfortunately no, without it `TypeListTest.cpp` doesn't compile on MSVC.